ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/metastore/events/AlterTableEvent.java

```diff
@@ -101,6 +112,36 @@ private HiveOperationType getOperationType() {
       ret.add(getHivePrivilegeObjectDfsUri(newUri));
     }
 
+    if(newTable.getParameters().containsKey(hive_metastoreConstants.META_TABLE_STORAGE)) {
+      String storageUri = "";
+      DefaultStorageHandler defaultStorageHandler = null;
+      HiveStorageHandler hiveStorageHandler = null;
+      Configuration conf = new Configuration();
+      Map<String, String> tableProperties = new HashMap<>();
+      tableProperties.putAll(newTable.getSd().getSerdeInfo().getParameters());
+      tableProperties.putAll(newTable.getParameters());
+      try {
+        hiveStorageHandler = (HiveStorageHandler) ReflectionUtils.newInstance(
+                conf.getClassByName(newTable.getParameters().get(hive_metastoreConstants.META_TABLE_STORAGE)), event.getHandler().getConf());
+        Method methodIsImplemented = hiveStorageHandler.getClass().getMethod("getURIForAuth", Map.class);
+        if(methodIsImplemented != null && hiveStorageHandler instanceof DefaultStorageHandler) {
+          DefaultStorageHandler defaultHandler = (DefaultStorageHandler) ReflectionUtils.newInstance(
+                  conf.getClassByName(newTable.getParameters().get(hive_metastoreConstants.META_TABLE_STORAGE)), event.getHandler().getConf());
+          storageUri = defaultHandler.getURIForAuth(tableProperties).toString();
+        }else if(methodIsImplemented != null && hiveStorageHandler instanceof HiveStorageAuthorizationHandler){
+          HiveStorageAuthorizationHandler authorizationHandler = (HiveStorageAuthorizationHandler) ReflectionUtils.newInstance(
+                  conf.getClassByName(newTable.getParameters().get(hive_metastoreConstants.META_TABLE_STORAGE)), event.getHandler().getConf());
+          storageUri = authorizationHandler.getURIForAuth(tableProperties).toString();
+        }
+      }catch(Exception ex){
+        //Custom storage handler that has not implemented the getURIForAuth()
+        storageUri = hiveStorageHandler.getClass().getName()+"://"+
```

Comment:
if above comment makes sense, then this should go into the else block of the condition instead of the exception handler.